File: python/muse/algorithms/sorting/MergeSortTest.py
from muse.algorithms.sorting import MergeSort
from muse.util import SequenceBuilder, Sequences, TestRunner


def testMergeSort() -> bool:
    size = 32768

    arr = [0] * size
    SequenceBuilder.packRandom(arr)

    checksum = Sequences.parityChecksum(arr)

    MergeSort.sort(arr)

    if Sequences.parityChecksum(arr) != checksum:
        return False

    return Sequences.isSorted(arr)


def main() -> None:
    TestRunner.pick(testMergeSort)


if __name__ == "__main__":
    main()

File: c++/muse/algorithms/sorting/MergeSortTest.cpp
#include <muse/algorithms/sorting/MergeSort.hpp>
#include <muse/util/SequenceBuilder.hpp>
#include <muse/util/Sequences.hpp>
#include <muse/util/TestRunner.hpp>

using namespace std;

bool testMergeSort() {
    size_t size = 32768;

    vector<int> arr(size);
    SequenceBuilder::packRandom(arr);

    int checksum = Sequences::parityChecksum(arr);

    MergeSort::sort(arr);

    if (Sequences::parityChecksum(arr) != checksum) {
        return false;
    }

    return Sequences::isSorted(arr);
}

int main() {
    TestRunner::pick(&testMergeSort);
}
